cmds/core/ip/ip.go

```diff
@@ -214,10 +218,131 @@ func routeshow() error {
 	if err != nil {
 		return fmt.Errorf("Route show failed: %v", err)
 	}
-	log.Printf("%s", string(b))
+	// Need to formation this better.
+	if !(*inet6) {
+	/*
+	ip: Iface	Destination	Gateway 	Flags	RefCnt	Use	Metric	Mask		MTU	Window	IRTT
+  ens33	00000000	02C210AC	0003	0	0	100	00000000	0	0	0
+  ens33	0000FEA9	00000000	0001	0	0	1000	0000FFFF	0	0	0
+  ens33	00C210AC	00000000	0001	0	0	100	00FFFFFF	0	0	0
+
+	want to turn it into something like this
+
+	eoinokane@ubuntu:~/.gvm/pkgsets/go1.12/global/src/github.com/u-root/u-root/cmds/core/ip$ ip route
+	default via 172.16.194.2 dev ens33 proto dhcp metric 100
+	169.254.0.0/16 dev ens33 scope link metric 1000
+	172.16.194.0/24 dev ens33 proto kernel scope link src 172.16.194.129 metric 100
+
+	*/
+		// Split the string by new line (Assumed format above)
+		rows := strings.Split(string(b),"\n")
+
+		// May be over kill but check for the format of the headerpattern for ubunto (is this this same
+		// for all linux implementations??? No clue, need to check with Andrea)
+		ubunto_headerpattern,_  := regexp.Compile("Iface	Destination	Gateway 	Flags	RefCnt	Use	Metric	Mask		MTU	Window	IRTT")
```

Comment:
This is a simple, documented format that won't change anytime soon. I'd rather split it instead of using regular expressions, see [strings.Fields](https://golang.org/pkg/strings/#Fields)


no need to reference ubuntu, this should be the same for other distributions

Reply:
then you can either use the titles to identify which field is which, or just use length verification and an index (I'd probably try the former, you can use that to validate the top line as a plus)